effort_controllers/src/joint_group_effort_controller.cpp

```diff
@@ -42,7 +42,7 @@ template <class T>
 void forward_command_controller::ForwardJointGroupCommandController<T>::starting(const ros::Time& time)
 {
   // Start controller with 0.0 efforts
-  commands_.resize(n_joints_, 0.0);
+  commands_buffer_.readFromRT()->assign(n_joints_, 0.0);
```

Comment:
You get no protection on the shared resource by doing this, because:

* You first get a pointer to the shared resource on `readFromRT()`, that's OK.
* You then write into the shared resource with no synchronization primitives in place.

The thing here is that `realtime_tools::RealtimeBuffer` does not handle well writing from real-time.

You can likely get away with the trick, because:
* The controller update in the `controller_manager` is serialized, and
* If all other writers to the shared resource (this would be controller ROS callbacks) use `writeFromNonRT`, they [will not write](https://github.com/ros-controls/realtime_tools/blob/indigo-devel/include/realtime_tools/realtime_buffer.h#L125) into the pointer [you're manipulating](https://github.com/ros-controls/realtime_tools/blob/indigo-devel/include/realtime_tools/realtime_buffer.h#L106).

So although kludgy, it will actually work.

The alternatives would be to either use `writeFromNonRT` and wait for the lock, or not use `realtime_tools::RealtimeBuffer` at all.